src/main/java/com/ea/restaurant/utils/Oauth2Util.java

```diff
@@ -56,15 +61,33 @@ public static String buildClientCredentialsToken(
 
   public static JWTClaimsSet getTokenDecoded(String token, String secretKey)
       throws BadJOSEException, ParseException, JOSEException {
-    ConfigurableJWTProcessor<SimpleSecurityContext> jwtProcessor =
-        new DefaultJWTProcessor<SimpleSecurityContext>();
-    JWKSource<SimpleSecurityContext> jweKeySource =
-        new ImmutableSecret<SimpleSecurityContext>(secretKey.getBytes());
-    JWEKeySelector<SimpleSecurityContext> jweKeySelector =
-        new JWEDecryptionKeySelector<SimpleSecurityContext>(
+    var jwtProcessor = buildJwtProcessor(secretKey);
+
+    return jwtProcessor.process(token, null);
+  }
+
+  public static void validateToken(String token, String secretKey)
+      throws BadJOSEException, ParseException, JOSEException {
+    getTokenDecoded(token, secretKey);
+  }
+
+  public static Integer getExpirationTimeInSeconds(Instant expirationTime) {
```

Comment:
Maybe you would like to use `iat` as well instead of `var now = Instant.now();`?

Reply:
no. In this case it is not what i want. because i don't want to get expiration time since token was create, instead i want to show remaining time from the moment of the refresh token method was called.

Reply:
What if you have 4 hours ahead from the timezone where the token was created? wouldn't this afec this logic?

Reply:
if i use `iat` the response will show a wrong expiration time as it will show expiration time since token was create even if there are 5 minutes left to expire.

Reply:
Got it!